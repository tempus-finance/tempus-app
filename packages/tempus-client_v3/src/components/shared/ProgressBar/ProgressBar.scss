$progress-bar-height: 12px;
$progress-bar-border: 1px;

.tc__progress-bar {
  width: 100%;
  height: $progress-bar-height;
  background-color: var(--progressBarBackground);
  border: $progress-bar-border solid var(--progressBarBorder);
  border-radius: calc($progress-bar-height / 2);
  box-sizing: border-box;

  .tc__progress-bar__progress {
    $progress-height: $progress-bar-height - 2 * $progress-bar-border;
    height: $progress-height;
    background-color: var(--progressBarProgressBackground);
    border-radius: calc($progress-height / 2);
    transition: width 300ms ease-out;
  }
}
