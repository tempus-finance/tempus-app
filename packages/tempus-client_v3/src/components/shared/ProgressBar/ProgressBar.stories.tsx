import { ComponentStory, ComponentMeta } from '@storybook/react';
import { CSSProperties } from 'react';
import ProgressBar from './ProgressBar';

export default {
  title: 'ProgressBar',
  component: ProgressBar,
  argTypes: {},
} as ComponentMeta<typeof ProgressBar>;

const style: CSSProperties = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  flexDirection: 'column',
  width: '300px',
  height: '180px',
};

const Template: ComponentStory<typeof ProgressBar> = () => (
  <div style={style}>
    {[...Array(6).keys()].map(index => (
      <ProgressBar value={20 * index} />
    ))}
  </div>
);

export const Primary = Template.bind({});
