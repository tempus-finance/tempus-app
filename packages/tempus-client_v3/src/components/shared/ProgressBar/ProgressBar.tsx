import { FC, memo } from 'react';
import './ProgressBar.scss';

export interface ProgressBarProps {
  value: number;
}

const ProgressBar: FC<ProgressBarProps> = props => {
  const { value } = props;

  return (
    <div className="tc__progress-bar">
      <div className="tc__progress-bar__progress" style={{ width: `${value}%` }} />
    </div>
  );
};

export default memo(ProgressBar);
