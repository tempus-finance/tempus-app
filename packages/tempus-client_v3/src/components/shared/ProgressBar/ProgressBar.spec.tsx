import { render } from '@testing-library/react';
import ProgressBar, { ProgressBarProps } from './ProgressBar';

const subject = (props: ProgressBarProps) => render(<ProgressBar {...props} />);

describe('ProgressBar', () => {
  it('renders a progress bar', () => {
    const { container } = subject({ value: 42 });

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
