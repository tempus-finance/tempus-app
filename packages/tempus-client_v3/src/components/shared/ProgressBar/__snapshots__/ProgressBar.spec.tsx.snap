// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders a progress bar 1`] = `
<div>
  <div
    class="tc__progress-bar"
  >
    <div
      class="tc__progress-bar__progress"
      style="width: 42%;"
    />
  </div>
</div>
`;
